### PR TITLE
[web] Implement SceneBuilder.pushColorFilter for html

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

Reuses Image/ColorFilter code to create layer with svg filter to implement ColorFilter.
Moved existing svg filter code to color_filter.dart.

## Related Issues

https://github.com/flutter/flutter/issues/61780

## Tests

Added color_filter_golden_test.dart.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

File: lib/web_ui/lib/src/engine/bitmap_canvas.dart [hunk 1]
```diff
@@ -573,57 +573,10 @@ class BitmapCanvas extends EngineCanvas {
...
-            _stringForBlendMode(colorFilterBlendMode));
-        break;
-      default:
-        break;
-    }
+    String? svgFilter = svgFilterFromBlendMode(filterColor,
+        colorFilterBlendMode);
     final html.Element filterElement =
-    html.Element.html(svgFilter, treeSanitizer: _NullTreeSanitizer());
+      html.Element.html(svgFilter, treeSanitizer: _NullTreeSanitizer());
```
Comment (hterkelsen) [inline lib/web_ui/lib/src/engine/bitmap_canvas.dart:579]: undo indent

File: lib/web_ui/lib/src/engine/html/color_filter.dart [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    if (colorFilterBlendMode != null) {
+      switch (colorFilterBlendMode) {
+        case ui.BlendMode.clear:
+        case ui.BlendMode.dstOut:
+        case ui.BlendMode.srcOut:
+          childContainer?.style.visibility = 'hidden';
+          return;
+        case ui.BlendMode.dst:
+        case ui.BlendMode.dstIn:
+          // Noop.
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/html/color_filter.dart:82]: Is it safe to keep the old value of `childContainer.style.visibility`? Should we set it to an empty string?

File: lib/web_ui/lib/src/engine/bitmap_canvas.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/bitmap_canvas.dart]: Done.

**Comment by ferhatb** [comment]
Merging on Fuschia infra fail.

**Comment by zanderso** [comment]
This is failing on CI with:
```
+ /b/s/w/ir/cache/builder/src/out/host_debug_unopt/dart-sdk/bin/dartanalyzer --enable-experiment=non-nullable --fatal-warnings --fatal-hints dev/ lib/ test/ tool/
Analyzing dev, lib, test, tool...
  warning â€¢ Missing case clause for 'darken'. â€¢ lib/src/engine/html/color_filter.dart:74:7 â€¢ missing_enum_constant_in_switch
1 warning found.
```

**Pull request merged**